plugin-server/src/cdp/consumers/cdp-events.consumer.ts

```diff
@@ -167,6 +168,49 @@ export class CdpEventsConsumer extends CdpConsumerBase {
                 'hog_function'
             )
 
+            const uniqueEventMetrics: MinimalAppMetric[] = [
+                ...new Set(notMaskedInvocations.map((invocation) => invocation.state.globals.event.uuid)),
+            ].reduce((acc, eventId) => {
+                const invocation = notMaskedInvocations.find(({ state }) => state.globals.event.uuid === eventId)
+                if (!invocation) {
+                    return acc
+                }
+                return [
+                    ...acc,
+                    {
+                        app_source: 'cdp_destinations',
+                        metric_kind: 'success',
+                        metric_name: 'event_triggered_destination',
+                        team_id: invocation!.state.globals.project.id,
+                        app_source_id: eventId,
+                        count: 1,
+                    },
+                ]
+            }, [] as MinimalAppMetric[])
+            this.hogFunctionMonitoringService.queueAppMetrics(uniqueEventMetrics, 'hog_function')
+
+            const uniqueDestinationMetrics: MinimalAppMetric[] = [
```

Comment:
Same applies to this one - you can even do the record building in the same loop so now you are doing one loop for the all metrics rather than potentially 6 loops.